crates/telio-dns/src/zone.rs

```diff
@@ -95,13 +105,17 @@ impl AuthoritativeZone {
             }
         }
 
-        Ok(AuthoritativeZone { zone })
+        Ok(AuthoritativeZone {
+            zone: Arc::new(zone),
+        })
     }
 }
 
 #[async_trait]
-impl Authority for AuthoritativeZone {
-    type Lookup = <InMemoryAuthority as Authority>::Lookup;
+impl AuthorityObject for AuthoritativeZone {
```

Comment:
`AuthorityObject` is already implemented for anything that implements `Authority` (and is inside of an `Arc`). Maybe instead of putting an `Arc` inside of our `AuthoritativeZone` we should just wrap it in an `Arc`?

Reply:
I fixed it. I hope I understand right. Please correct me if not.